INSTALL.md

```diff
@@ -102,7 +102,7 @@ for its options, which include:
     blas_threaded
         Whether to search for multi-threaded or sequential BLAS.
         Currently applies to Intel MKL and IBM ESSL. One of:
-        1 (yes)         multi-threaded BLAS
+        1 (yes)         multi-threaded BLAS (set for Intel-OneMKL (SYCL))
```

Comment:
"Also update INSTALL.md to change blas_threaded default."
In CMakeLists.txt, blas_threaded is set to "auto" and obtained from BLAS++.
In GNUmakefile, blas_threaded is set to 1 for SYCL/OneMKL, 0 otherwise.
This comment seems to be a way to express that.


"Note that I don't think we can use multi-threaded ESSL, or at least it isn't tested. While MKL recognizes when it is inside an OpenMP parallel region, I don't know that ESSL does that. So maybe we need a smarter blas_threaded option that is auto for SYCL and false otherwise."
I put a note in the CMakeLists.txt to say that the BLAS needs to OpenMP-aware.

Reply:
"I think it would be better to change the CMake file so it matches the Makefile, namely to set threaded = true if gpu_backend = sycl, and otherwise leave threaded = false by default "

I had considered this.  But, the blas search (using blas_threaded) needs to be done before the gpu_backend (specifically rocm) search.
```
# Search for BLAS library, if not already included.
# Needed before ROCm, for include paths.
```
As far as I can tell, checking `gpu_backend=sycl` before setting `blas_threaded` requires restructuring the CMakefile.txt.   I could move the check for SYCL support before the BLAS options section (separated from the other gpu_backend probes).  Unless you have a better idea.

Reply:
I have put a commit that moves SYCL search before the BLAS options in CMakelists.txt
I don't think this disturbs the structure of CMakelists too much.  [5a48114]

I corrected the OneMKL to oneMKL in [5dab438].   Sorry, I will try to be careful of that.

Reply:
If [5a48114] is OK, then the comments in the INSTALL.md should have the Makefile and the CMake at the same point.  If SYCL+oneMKL, then use blas_threaded=true, else use blas_threaded=false.

Reply:
Let's test ESSL to see if it's a problem.

I think ROCm discovery can happen before BLAS++, it's the hipify step that has to happen after BLAS++ because the *.cu files indirectly include BLAS++ headers.